syntax = "proto3";

message CreatePersonRequest {
    string first_name = 1;
    string last_name = 2;
    string company_name = 3;
}

message GetPersonRequest {
    int32 person_id = 1;
}

message ListPersonsRequest {
}

message Person {
    int32 id = 1;
    string first_name = 2;
    string last_name = 3;
    string company_name = 4;
}

message ListPersonsResponse {
    repeated Person persons = 1;
}

service PersonService {
    rpc CreatePerson(CreatePersonRequest) returns (Person);
    rpc GetPerson(GetPersonRequest) returns (Person);
    rpc ListPersons(ListPersonsRequest) returns (ListPersonsResponse);
}
